### ISSUE TITLE
NULL pointer dereference in of

### ISSUE THREAD

**Issue by gy741** [kevinlawler/kona#491]
Hello.

I found a NULL pointer dereference bug in kona.

Please confirm.

Thanks.

Summary: NULL pointer dereference
OS: Ubuntu 16_04 64bit
Version: commit cd1365744b1abf0cd8247f394b26d0ad34fcae39
PoC Download: [Null_of.zip](https://github.com/kevinlawler/kona/files/1717781/Null_of.zip)

Steps to reproduce:
1.Execute the following command
: ./k $FILE

```
=================================================================
==7518==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x000000576658 bp 0x7ffd7a82a4d0 sp 0x7ffd7a82a1c0 T0)
    #0 0x576657 in of /home/karas/kona/src/vd.c:69:17
    #1 0x577dfc in dot /home/karas/kona/src/vd.c:121:5
    #2 0x53a205 in vf_ex /home/karas/kona/src/kx.c:453:71
    #3 0x5339eb in dv_ex /home/karas/kona/src/kx.c:399:16
    #4 0x540b16 in ex2 /home/karas/kona/src/kx.c:1033:5
    #5 0x53f6b5 in ex2 /home/karas/kona/src/kx.c:999:6
    #6 0x546224 in ex1 /home/karas/kona/src/kx.c:830:35
    #7 0x536cff in ex0 /home/karas/kona/src/kx.c:697:5
    #8 0x545657 in ex_ /home/karas/kona/src/kx.c:671:5
    #9 0x544fb2 in ex /home/karas/kona/src/kx.c:680:7
    #10 0x52b5e3 in line /home/karas/kona/src/kc.c:287:3
    #11 0x52d4d9 in lines /home/karas/kona/src/kc.c:250:41
    #12 0x4f7a4c in load /home/karas/kona/src/c.c:85:3
    #13 0x5285aa in args /home/karas/kona/src/kc.c:93:21
    #14 0x592c48 in main /home/karas/kona/src/main.c:7:3
    #15 0x7feaa698982f in __libc_start_main /build/glibc-Cl5G7W/glibc-2.23/csu/../csu/libc-start.c:291
    #16 0x43a238 in _start (/home/karas/kona/k+0x43a238)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/karas/kona/src/vd.c:69 of
==7518==ABORTING
```

==========
[Acknowledgement]
This work was supported by ICT R&D program of MSIP/IITP. [R7518-16-1001, Innovation hub for high Performance Computing]

**Comment by tavmem** [comment]
Though the fix on this issue avoided the seg fault, the result was not fully compatible.
First, of all, what's in Null_of
```
$ rlwrap -n ~/k2.8/k
K 2.8 2000-10-10 Copyright (C) 1993-2000 Kx Systems
Evaluation. Not for commercial use. 
\ for help. \\ to exit.
  0: "Null_of"
(".n"
 "@`\"\377\377\377\177#\"\"\023\"")
  \\ 
```
Executing the first line yields "value error" in both k2.8 and kona.
However, the second line yields differing results.
In k2.8
```
$ rlwrap -n ~/k2.8/k
K 2.8 2000-10-10 Copyright (C) 1993-2000 Kx Systems
Evaluation. Not for commercial use. 
\ for help. \\ to exit.
  .( 0: "Null_of")[1]
domain error
```
In kona
```
$ rlwrap -n ~/kona/k
K Console - Enter \ for help
  .(0:"Null_of")[1]
syntax error
```
Kona should be reporting a "domain error" instead of a "syntax error".

**Comment by tavmem** [comment]
This is fixed in commit 762f5267d4b17dbeda63036620cdaad513e84556
